engine/src/main/scala/cromwell/services/ServiceRegistryActor.scala

```diff
@@ -0,0 +1,50 @@
+package cromwell.services
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.{ConfigFactory, Config, ConfigObject}
+import cromwell.engine.db.DataAccess
+import cromwell.services.ServiceRegistryActor._
+import lenthall.config.ScalaConfig._
+
+import scala.collection.JavaConverters._
+
+object ServiceRegistryActor {
+  case class ServiceRegistryFailure(serviceName: String)
+  trait ServiceRegistryMessage {
+    def serviceName: String
+  }
+
+  def props(dataAccess: DataAccess, config: Config) = {
+    Props(ServiceRegistryActor(dataAccess, config))
+  }
+}
+
+case class ServiceRegistryActor(dataAccess: DataAccess, globalConfig: Config) extends Actor {
+  import ServiceRegistryActor._
+
+  val services: Map[String, ActorRef] = {
+    val serviceNamesToConfigStanzas = globalConfig.getObject("services").entrySet.asScala.map(x => x.getKey -> x.getValue).toMap
+    serviceNamesToConfigStanzas map {
+      case (serviceName, config: ConfigObject) => serviceName -> makeActor(serviceName, config.toConfig)
+      case (serviceName, _) => throw new Exception(s"invalid configuration for service $serviceName")
+    }
+  }
+
+  private def makeActor(serviceName: String, serviceStanza: Config): ActorRef = {
+    val serviceConfigStanza = serviceStanza.getConfigOption("config").getOrElse(ConfigFactory.parseString(""))
+    val className = serviceStanza.getStringOption("class").getOrElse(
+      throw new Exception(s"invalid configuration for service $serviceName: missing 'class' definition")
```

Comment:
`.getStringOption(a).getOrElse(b)` should be `.getStringOr(a, b)`. Not sure why lenthall doesn't have a `.getConfigOr`.

Also, "invalid" -> "Invalid"